### ISSUE TITLE
<select>s in tall Modals in iOS 8.3 break fixed alignment and scrolling

### ISSUE THREAD

**Issue by SimonKaluza** [twbs/bootstrap#16557]
Whenever you deselect/blur on a `<select>` dropdown that is within a Bootstrap Modal that has vertical overflow/scrollbars in iOS 8.x Safari or iOS8's Webkit WebView, [Apple's `position: fixed` bug](http://getbootstrap.com/getting-started/#support-fixed-position-keyboards) badly messes up the alignment of any other HTML element within the modal.  The browser will believe any of the HTML elements in the modal are at a very different location than what is actually rendered, and this can be verified by examining the page using the Web Inspector.

Checkout this [JSBin example](http://jsbin.com/zizaba/3) on an iOS device (preferably within the iPad Simulator with Web Inspector) and tab to the dropdown at the bottom, select an item, then try to select one of the text inputs within the modal.  You'll see that in order to select an input, you'll have to click/touch where the browser *thinks* the `<input>` element is, not where it actually appears on screen.  To make matters worse, after removing focus from the `<select>` element, you won't be able to scroll the modal up or down at all until you select another input element within the modal.

Here is a screenshot demonstrating how Safari believes the modal-content div is positioned higher (and where it responds to mouse/touch events) as opposed to where is actually rendered:

![modal-dropdown-bug](https://cloud.githubusercontent.com/assets/648386/7872179/1b66b942-0564-11e5-8194-1245b9648c72.png)

I've come up with a hack to fix the `position: fixed` alignment bugs by adding a listener to any `select` element's `blur` event that realigns the content by forcing the HTML body to scroll to the top of the page with the following code:

    $('modal').find('select').blur(function(){
        $('body').scrollTop(0);
    });

But this hack does not fix the modals scrolling even after the alignment is fixed.  Any ideas on how one might fix that?

**Comment by twbs-lmvtfy** [comment]
Hi @SimonKaluza!

You appear to have posted a live example (http://jsbin.com/zizaba/3/edit), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 25, column 3 thru line 25, column 117: The ```` aria-labelledby ```` attribute must point to an element in the same document.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by SimonKaluza** [comment]
You got it @twbs-lmvtfy bot, though that validation error was just a remnant from some boilerplate copy/paste and doesn't seem to be causing the issue.
http://jsbin.com/zizaba/4

**Comment by twbs-lmvtfy** [comment]
Hi @SimonKaluza!

You appear to have posted a live example (http://jsbin.com/zizaba/4/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 32, column 5: [E042](https://github.com/twbs/bootlint/wiki/E042): `.form-control` cannot be used on non-textual `<input>`s, such as those whose `type` is: `file`, `checkbox`, `radio`, `range`, `button`
* line 36, column 5: [E042](https://github.com/twbs/bootlint/wiki/E042): `.form-control` cannot be used on non-textual `<input>`s, such as those whose `type` is: `file`, `checkbox`, `radio`, `range`, `button`
* line 40, column 5: [E042](https://github.com/twbs/bootlint/wiki/E042): `.form-control` cannot be used on non-textual `<input>`s, such as those whose `type` is: `file`, `checkbox`, `radio`, `range`, `button`
* line 44, column 5: [E042](https://github.com/twbs/bootlint/wiki/E042): `.form-control` cannot be used on non-textual `<input>`s, such as those whose `type` is: `file`, `checkbox`, `radio`, `range`, `button`
* line 48, column 5: [E042](https://github.com/twbs/bootlint/wiki/E042): `.form-control` cannot be used on non-textual `<input>`s, such as those whose `type` is: `file`, `checkbox`, `radio`, `range`, `button`
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* line 7, column 3: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.4 ; saw what appears to be usage of Bootstrap 3.3.2
* line 8, column 3: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.4 ; saw what appears to be usage of Bootstrap 3.3.2

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by SimonKaluza** [comment]
Alright, how bout now http://jsbin.com/zizaba/4

**Comment by cvrebert** [comment]
I was able to repro against current `master` in iOS 8.3 with an iPhone 5S using http://jsbin.com/litazi/2/edit

**Comment by twbs-lmvtfy** [comment]
Hi @cvrebert!

You appear to have posted a live example (http://jsbin.com/litazi/2/edit), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 22, column 3 thru line 22, column 117: The ```` aria-labelledby ```` attribute must point to an element in the same document.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by SimonKaluza** [comment]
Though the broken scrolling issue is present on both iPhone and iPad devices, it seems the content alignment issue only occurs on iPads (confirmed on iPad Air and iPad 2 hardware and simulator).

**Comment by max-tubecorporate** [comment]
Try this http://jsbin.com/fulecagami/1/edit

**Comment by twbs-lmvtfy** [comment]
Hi @max-tubecorporate!

You appear to have posted a live example (http://jsbin.com/fulecagami/1/edit), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 32, column 3 thru line 32, column 117: The ```` aria-labelledby ```` attribute must point to an element in the same document.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
@max-tubecorporate Could you summarize what you changed in your version?

**Comment by MaxKorz** [comment]
@cvrebert sorry I used wrong account.

http://jsbin.com/fulecagami/1/edit

I added
```html
  <style>
    .modal-open {
      position: fixed;
      width: 100%;
      height: 100%;
      overflow: hidden;
    }
  </style>
```

**Comment by twbs-lmvtfy** [comment]
Hi @MaxKorz!

You appear to have posted a live example (http://jsbin.com/fulecagami/1/edit), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 32, column 3 thru line 32, column 117: The ```` aria-labelledby ```` attribute must point to an element in the same document.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by SimonKaluza** [comment]
@MaxKorz Those 4 properties totally fixed it _and_ seem to still work great in the other browsers I've manually tested.  Very nice!

**Comment by cvrebert** [comment]
Perhaps related to (or duplicate of) #14975?

**Comment by bkingg** [comment]
@MaxKorz Thanks. It fixed it for me as well

**Comment by beedaan** [comment]
Thank you @MaxKorz.  Your solution fixed my problem.

**Comment by cvrebert** [comment]
Duplicate of #14975, AFAICT.
We've filed a bug against WebKit: https://bugs.webkit.org/show_bug.cgi?id=150079

**Comment by jesse-peekabuy** [comment]
@MaxKorz Thank you, worked for me too.

**Comment by beedaan** [comment]
Just to follow up, @MaxKorz's solution sort of fixes the problem by introducing reliable behavior, though it may not be intentional.  The page will scroll to the top when the modal is open, regardless of browser.  This is not a perfect fix for me because it breaks the behavior in desktop browsers.

I also found that the problem exists for the `<input>` field.  I know there has been a bug filed for WebKit.  I think it important to keep all interested parties up to date on this issue.

Updated jsbin with `<input>` http://jsbin.com/nugihe

**Comment by twbs-lmvtfy** [comment]
Hi @beedaan!

You appear to have posted a live example (http://jsbin.com/nugihe/edit), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 88, column 17 thru column 54: Attribute ```` inputmode ```` not allowed on element ```` input ```` at this point.
* line 89, column 17 thru column 24: Stray end tag ```` input ````.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by beedaan** [comment]
Just to follow up, @MaxKorz's solution sort of fixes the problem by introducing reliable behavior, though it may not be intentional.  The page will scroll to the top when the modal is open, regardless of browser.  This is not a perfect fix for me because it breaks the behavior in desktop browsers.

I also found that the problem exists for the `<input>` field.  I know there has been a bug filed for WebKit.  I think it important to keep all interested parties up to date on this issue.

Updated jsbin with `<input>` http://jsbin.com/nugihe

**Comment by twbs-lmvtfy** [comment]
Hi @beedaan!

You appear to have posted a live example (http://jsbin.com/nugihe/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 61, column 13: [E003](https://github.com/twbs/bootlint/wiki/E003): Found one or more `.row`s that were not children of a grid column or descendants of a `.container` or `.container-fluid`
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* line 24, column 12: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 66, column 17: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 8, column 5: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.6 ; saw what appears to be usage of Bootstrap 3.3.0
* line 100, column 5: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.6 ; saw what appears to be usage of Bootstrap 3.3.0

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by GrayYoung** [comment]
The solution of @MaxKorz is perfect.
